### ISSUE TITLE
(SUGGESTION) Apply Theme to Bookmarks Subpanel

### ISSUE THREAD

**Issue by Nano-Ocelot** [piroor/tst-bookmarks-subpanel#32]
When using the bookmarks subpanel add-on, the theme for it isn't matching the rest of the sidebar or even the default bookmarks sidebar for that matter. I'm not sure if this is a bug or not but I think it would look much better if they matched.

![Screenshot_20210228_210448](https://user-images.githubusercontent.com/34014750/109447109-f7c6d300-7a10-11eb-8271-1377422f0e4c.png)
Notice the top and bottom part of the sidebar are different colors.

## Steps to reproduce

 1. Install both extensions with something other than the default firefox theme active. I'm also using firefox CSS theming if that matters.
 
2. Notice that the bookmarks subpanel (F2) isn't the same color as the rest of the sidebar (colors set in CSS for the base add-on) or the default bookmarks sidebar (the colors set by CSS in my firefox profile folder)

## Expected result

One would expect the theme to match up

## Actual result

It doesn't match up with either theming source and I'm not sure how to get them to match up.

## Environment

 * Platform (OS): Linux - OpenSUSE Tumbleweed
 * Version of Firefox: 85.0.1
 * Version (or revision) of Tree Style Tab: 3.7.2

**Comment by piroor** [comment]
This is a topic not about TST itself but covered by the TST Bookmarks Subpanel, thus I've transferred this to https://github.com/piroor/tst-bookmarks-subpanel/issues

**Comment by piroor** [comment]
Thanks, it was unexpected and I've introduced some changes to match the appearance of the search field to Dark UI more. Here is a screenshot on Ubuntu 20.04LTS + Yaru-dark:
![image](https://user-images.githubusercontent.com/70062/109514116-054f8c00-7ae9-11eb-8f2d-bc6ac2f552d7.png)

**Comment by Nano-Ocelot** [comment]
The search bar is much better! Awesome. What about the rest of the subpanel though? Is there any hope of it respecting perhaps either the colors of the default bookmarks sidebar or even the colors set within the TST main add-on? Or even a possibility to manually css it?

**Comment by piroor** [comment]
Currently both TST and TST Bookmarks Subpanel support only Firefox's Dark color scheme. And TST Bookmarks Subpanel currently respects the color scheme of in-content UI of Firefox (e,g, "about:addons") instead of Firefox's chrome UI.

For example, here is a screenshot with the GTK theme "Adapta-Nokto":
![image](https://user-images.githubusercontent.com/70062/109996011-2830a900-7d52-11eb-8df7-f1721ffcbe4a.png)
"Adapta-Nokto" provides dark-blue colors, but as you saw Firefox's in-content UI "about:addons" (moreover TST and TST Bookmarks Subpanel) has different dark color.

Sadly, colors from GTK won't be applied to addon contents by default for security reasons (to prevent fingerprinting). TST's "High Contrast" theme uses CSS2 system colors (and more Firefox specific system color names) but actually TST's contents are rendered with fake system colors like as:
![image](https://user-images.githubusercontent.com/70062/109996686-cf154500-7d52-11eb-9c86-0014d65b8f92.png)

If you turn `widget.content.allow-gtk-dark-theme` to `true` via "about:config", GTK theme colors are applied to TST's sidebar. Here is a screenshot:
![image](https://user-images.githubusercontent.com/70062/109997105-37642680-7d53-11eb-9c61-3046c38a6b19.png)
But TST Bookmarks Subpanel still uses different color: we need to override color definitions in TST Bookmarks Subpanel with system colors.

This means that we need add a new option to switch color scheme of TST Bookmarks Subpanel between the default colors (matching to Firefox's Light and Dark themes) and the system colors.
I'm very sorry but I have no plan to do that by me for now. Reasons:

* The color scheme with system colors is fragile. It requires more costs to maintain that.
* I generally use Windows 10, so TST's "dogfooding" is ran on Windows. Frequent testings on Linux desktop environment also require more costs.
* I ordinaly don't use TST Bookmarks Subpanel on my environment, I'm providing TST Bookmarks Subpanel just as a demonstration of TST's subpanel API. Thus I have very limited motivation to maintain this.

In other words, contribution from users is welcome. Adding system color scheme support looks a good first topic for contribution, because there are hints in TST itself.

**Comment by Nano-Ocelot** [comment]
Ah, I was hoping for a css theming option like the main add-on has. I understand if that's too intensive to implement and maintain. Either way thanks for the hard work you've put in to a great add-on!
